### PR TITLE
feat: puppyos

### PR DESCRIPTION
**Pull request by cappsyco**
A litter of puppies!

Included are all the official versions. The non-'debian' distros are not described as variants in their files so they have their own folders for that reason.

Second note is the two versions in the 'puppy' folder are the 32 and 64 bit versions for the Debian variant. They both have 'debian' as ID but 'Puppy' as NAME so I put them in the same folder as they have different version numbers. Is it possible I should've used their pretty names for their folders in this case?

Definitely a weird distro all round.

### CODE REVIEW

**Comment by SteveLauC** [comment]
> The non-'debian' distros are not described as variants in their files so they have their own folders for that reason.

Using their `ID` as the directory name makes sense:)


> Second note is the two versions in the 'puppy' folder are the 32 and 64 bit versions for the Debian variant. They both have 'debian' as ID but 'Puppy'

Using `puppy` looks good:

> unless it does not have its own unique ID,
> 
> in this case, we turn its `NAME` into lowercase, replace non-alphabetic characters with _, and use this as the directory name.

File: puppy_fossapup64/puppy_fossapup64
Comment (SteveLauC) [inline puppy_fossapup64/puppy_fossapup64:1]: For this and the following 2 distros, they are based on `Ubuntu` and `Slackware`, which use fixed release mode, so we should use the `VERSION_ID` as the file name? 🤔

Comment (cappsyco) [reply puppy_fossapup64/puppy_fossapup64:1]: Oh whoops! Fixed that now.

**Review by SteveLauC — APPROVED** [review]
Thanks

**Pull request merged**